pkg/resources/secondary_database.go

```diff
@@ -32,36 +32,15 @@ var secondaryDatabaseSchema = map[string]*schema.Schema{
 		ForceNew:    true,
 		Description: "Specifies the database as transient. Transient databases do not have a Fail-safe period so they do not incur additional storage costs once they leave Time Travel; however, this means they are also not protected by Fail-safe in the event of a data loss.",
 	},
-	"data_retention_time_in_days": {
-		Type:     schema.TypeList,
-		MaxItems: 1,
-		Elem: &schema.Resource{
-			Schema: map[string]*schema.Schema{
-				"value": {
-					Type:     schema.TypeInt,
-					Required: true,
-				},
-			},
-		},
-		Computed:    true,
-		Optional:    true,
-		Description: "Specifies the number of days for which Time Travel actions (CLONE and UNDROP) can be performed on the database, as well as specifying the default Time Travel retention time for all schemas created in the database. For more details, see [Understanding & Using Time Travel](https://docs.snowflake.com/en/user-guide/data-time-travel).",
-	},
-	"max_data_extension_time_in_days": {
-		Type:     schema.TypeList,
-		MaxItems: 1,
-		Elem: &schema.Resource{
-			Schema: map[string]*schema.Schema{
-				"value": {
-					Type:     schema.TypeInt,
-					Required: true,
-				},
-			},
-		},
-		Computed:    true,
-		Optional:    true,
-		Description: "Object parameter that specifies the maximum number of days for which Snowflake can extend the data retention period for tables in the database to prevent streams on the tables from becoming stale. For a detailed description of this parameter, see [MAX_DATA_EXTENSION_TIME_IN_DAYS](https://docs.snowflake.com/en/sql-reference/parameters.html#label-max-data-extension-time-in-days).",
-	},
+	"data_retention_time_in_days": nestedProperty(
+		schema.TypeInt,
+		"Specifies the number of days for which Time Travel actions (CLONE and UNDROP) can be performed on the database, as well as specifying the default Time Travel retention time for all schemas created in the database. For more details, see [Understanding & Using Time Travel](https://docs.snowflake.com/en/user-guide/data-time-travel).",
+	),
+	"max_data_extension_time_in_days": nestedProperty(
+		schema.TypeInt,
+		"Object parameter that specifies the maximum number of days for which Snowflake can extend the data retention period for tables in the database to prevent streams on the tables from becoming stale. For a detailed description of this parameter, see [MAX_DATA_EXTENSION_TIME_IN_DAYS](https://docs.snowflake.com/en/sql-reference/parameters.html#label-max-data-extension-time-in-days).",
+	),
+	// TODO: Below parameters should be nested properties
```

Comment:
A thing I found out about, will be a part of the next pr.

Reply:
oky